infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/ErrorState.java

```diff
@@ -24,8 +24,6 @@ public LogReplicationState processEvent(LogReplicationEvent event) {
 
     @Override
     public void onEntry(LogReplicationState from) {
-        // Disable periodic sync status periodic task while in initialized state (no actual replication occurring)
-        fsm.getAckReader().stopSyncStatusUpdatePeriodicTask();
```

Comment:
when the fsm reaches Initialized state, it schedules the ack polling task with fixed delay perpetually.  If it is not stopped, the task will keep running and can lead to inconsistent results.

One way to address this can be to add a set of error fsms/sessions in AckReader and add this session.  Until the session is removed from that set, the polling task will be a no-op.

Reply:
I found that cancelling the future removes the task from the pool's workQueue. So have taken that approach